### PR TITLE
Remove duplicate Solidity sig check in fullnode proxy

### PR DESCRIPTION
**Pull request by karlfloersch**
## Description
This PR speeds up tests by no longer using the fully Solidity validated EOA function call. Instead we just recover the tx contents once inside of the fullnode, and then make an authenticated call on that user's behalf.

**It also** fixes a _really_ tricky bug which was causing our fullnode to crash **inside of a `try{}catch{}` statement!!**. Apparently you cannot use Ganache's `call` with the ZERO_ADDRESS if the call fails because that will cause a really nasty error:

```
/Users/karlfloersch/workspace/ethereum/optimism-monorepo/node_modules/ethereumjs-util/dist/index.js:369
  var sig = secp256k1.sign(msgHash, privateKey);
                      ^
Error: nonce generation function failed or private key is invalid
    at Object.exports.ecsign (/Users/karlfloersch/workspace/ethereum/optimism-monorepo/node_modules/ethereumjs-util/dist/index.js:369:23)
    at Transaction.sign (/Users/karlfloersch/workspace/ethereum/optimism-monorepo/node_modules/ethereumjs-tx/es5/index.js:252:23)
```

By making this small change: https://github.com/ethereum-optimism/optimism-monorepo/pull/23/commits/b4060c8979f5a909aa76c5e21e61aff9f39031bd it fixed the issue. Something that's a little weirder is that doing something as simple as changing it to use `0x 0x0000000000000000000000000000000000000099` as the `from` field also solves the issue! So it's just something special about the ZERO_ADDRESS it seems...

## TODO
- [x] It is probably best to verify that the nonce is correct. This should be a quick fix.

### Fixes
- Fixes # [YAS-194](https://optimists.atlassian.net/browse/YAS-194?atlOrigin=eyJpIjoiYjgwZTAxYTQyMDk4NGYyNDhhMWE4YmYxZWY3ZDhiMzciLCJwIjoiaiJ9)

## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](https://github.com/ethereum-optimism/optimism-monorepo/blob/master/.github/CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

File: packages/rollup-full-node/src/app/handler.ts [hunk 1]
```diff
@@ -445,16 +445,14 @@ export class DefaultWeb3Handler implements Web3Handler, FullnodeHandler {
     // Generate the calldata which we'll use to call our internal execution manager
     // First pull out the `to` field (we just need to check if it's null & if so set ovmTo to the zero address as that's how we deploy contracts)
     const ovmTo = ovmTx.to === null ? ZERO_ADDRESS : ovmTx.to
+    // TODO: Verify that the nonce on the transaction is correct
     // Construct the raw transaction calldata
-    const internalCalldata = this.generateEOACallCalldata(
+    const internalCalldata = this.generateUnsignedCallCalldata(
```
Comment (willmeister) [inline packages/rollup-full-node/src/app/handler.ts]: 👍 

Would change to `getCallData(...)` and `getTransactionData(...)` to better indicate how these two are used in the grand scheme of things.

File: packages/ovm/src/contracts/ExecutionManager.sol [hunk 2]
```diff
@@ -504,12 +502,13 @@ contract ExecutionManager is FullStateManager {
      * @param _timestamp The timestamp which should be used for this context.
      * @param _queueOrigin The queue which this context's transaction was sent from.
```
Comment (K-Ho) [inline packages/ovm/src/contracts/ExecutionManager.sol]: Add an @param comment for _ovmTxOrigin

File: packages/ovm/test/helpers.ts [hunk 3]
```diff
@@ -105,7 +106,8 @@ export const executeUnsignedEOACall = async (
   executionManager: Contract,
   wallet: Wallet,
   to: Address,
-  data: string
+  data: string,
+  allowRevert: boolean
 ): Promise<TransactionReceipt> => {
```
Comment (K-Ho) [inline packages/ovm/test/helpers.ts]: A function comment for `executeUnsignedEOACall` describing what each of the parameters is would be helpful! (Specifically about allowRevert)

File: packages/rollup-full-node/src/app/web3-rpc-handler.ts [hunk 4]
```diff
@@ -524,50 +533,27 @@ export class DefaultWeb3Handler implements Web3Handler, FullnodeHandler {
     return executionManager
   }
 
-  private generateUnsignedCallCalldata(
+  private getTransactionCalldata(
```
Comment (K-Ho) [inline packages/rollup-full-node/src/app/web3-rpc-handler.ts]: A short function comment would be helpful!

**Pull request merged**